pyfstat/injection_parameters.py

```diff
@@ -1,83 +1,256 @@
-        self.priors = self._parse_priors(self.priors)
+        self._rng = generator or np.random.default_rng(seed)
 
-    def _parse_priors(self, priors_input_format):
+    def _deprecated_prior_parsing(self, parameter_prior) -> dict:
+        # FIXME: Will be removed in a future release
+        rng_function_name = next(iter(parameter_prior))
+        rng_function = getattr(self._rng, rng_function_name)
+        rng_kwargs = parameter_prior[rng_function_name]
+        return functools.partial(rng_function, **rng_kwargs)
+
+    def _parse_priors(self, priors_input_format: dict):
         """Internal method to do the actual prior setup."""
-        priors = {}
+        self.priors = {}
 
         for parameter_name, parameter_prior in priors_input_format.items():
+
             if callable(parameter_prior):
-                priors[parameter_name] = parameter_prior
-            elif isinstance(parameter_prior, dict):
-                rng_function_name = next(iter(parameter_prior))
-                rng_function = getattr(self._rng, rng_function_name)
-                rng_kwargs = parameter_prior[rng_function_name]
-                priors[parameter_name] = functools.partial(rng_function, **rng_kwargs)
-            else:  # Assume it is something to be returned as is
-                priors[parameter_name] = functools.partial(lambda x: x, parameter_prior)
-
-        return priors
-
-    def draw(self):
+                # FIXME Deprecated, to be removed on a future release.
+                logger.warning(
+                    f"Parameter `{parameter_name}`'s prior was specified as a callable, "
+                    "which is not covered by the current implementation of RNGs "
+                    "and hence will not make use of the specified seed. "
+                    "Please, beware of the new implementation of parameter priors. "
+                    "This will raise an error in the future."
+                )
+                self.priors[parameter_name] = parameter_prior
+                continue
+
+            if not isinstance(parameter_prior, Mapping):
+                # If not dictionary, then return as is
+                self.priors[parameter_name] = lambda val=parameter_prior: val
+                continue
+
+            distribution = next(iter(parameter_prior))
+
+            if distribution in _pyfstat_custom_priors:
+                logger.debug(
+                    f"Setting {distribution} distribution form `_pyfstat_custom_priors` "
+                    f"with kwargs `{parameter_prior[distribution]}` "
+                    f"to parameter {parameter_name}"
+                )
+                custom_function = _pyfstat_custom_priors[distribution]
+                self.priors[parameter_name] = functools.partial(
+                    custom_function,
+                    generator=self._rng,
+                    **parameter_prior[distribution],
+                )
+            elif "stats." in distribution:
```

Comment:
check

- that it's a string at this point
- that it's not a custom prior of `x.stats.y` format...?

Reply:
Point 1) done, point 2) is illegal in python and is not allowed by `custom_priors` either, as discussed offline.